### PR TITLE
Disable client side validation

### LINKED ISSUE #243: AutoRest should provide a way to disable client-side validation
Flag doesn't work. If we are using autorest-typescript for building API test's, we don't want the client-side validation included.

### PR DESCRIPTION
**Pull request by RikkiGibson**
Resolves #243 by simply not emitting the constraints.

We may wish to generate additional tsdoc that describes the constraints that we check at runtime, but if we do that it will be part of a separate task.

### CODE REVIEW

File: src/vanilla/ClientModelExtensions.cs [hunk 1]
```diff
@@ -727,7 +728,8 @@ public static void ConstructMapper(TSObject mapper, IModelType type, string seri
 
             void applyConstraints(TSObject obj)
             {
-                if (constraints != null && constraints.Any())
+                bool useClientSideValidation = (bool) Settings.Instance.CustomSettings["ClientSideValidation"];
```
Comment (daschult) [inline src/vanilla/ClientModelExtensions.cs]: Can you add this to GeneratorSettingsTS? That way we don't have to use the string literal.

**Comment by RikkiGibson** [comment]
325cd87 shows the difference

**Comment by shabfactor** [comment]
@RikkiGibson  so do i need to install latest version to make it work for TS?

**Comment by RikkiGibson** [comment]
Yes, please try generating with `@microsoft.azure/autorest.typescript@preview` and install `ms-rest-js@preview` in your project

**Comment by shabfactor** [comment]
@RikkiGibson  when i tried to install `ms-rest-js@preview` it did ask for the version to install, which one i should be installing?

**Comment by RikkiGibson** [comment]
Please try 0.19.380

**Comment by DavidNorena** [comment]
is this documented ?

**Pull request merged**